app/code/core/Mage/Core/Model/Cache.php

```diff
@@ -336,65 +190,55 @@ protected function _tags($tags = [])
         return $tags;
     }
 
-    /**
-     * Get cache frontend API object
-     *
-     * @return Varien_Cache_Core|Zend_Cache_Core
-     */
-    public function getFrontend()
+    // @phpstan-ignore return.internalClass
+    public function getFrontend(): \Symfony\Component\Cache\Adapter\AbstractTagAwareAdapter
     {
         return $this->_frontend;
     }
 
     /**
      * Load data from cache by id
-     *
-     * @param   string $id
-     * @return  string|false
      */
-    public function load($id)
+    public function load(string $id): mixed
     {
-        return $this->getFrontend()->load($this->_id($id));
+        $item = $this->getFrontend()->getItem($this->_id($id)); // @phpstan-ignore method.internalClass
+        if ($item->isHit()) {
+            return $item->get();
+        }
+        return false;
```

Comment:
this (and the return false) is done to maintain the previous behavior, I've tested not checking for isHit() but all of the framework broke so...

Reply:
It doesn't look like `isHit` has very much overhead ([ref](https://github.com/symfony/symfony/blob/11495f445f0ffb8f55981d2a176b5f0525fcdf53/src/Symfony/Component/Cache/CacheItem.php#L48-L51)), but could this work too?

```php
return $item->get() ?? false;
```

Reply:
@justinbeaty would it make sense to store `null` in cache? cause if the cache is `hit` but null, I'd still return null 🤔

Reply:
Yes, you're correct and I hadn't thought of that when I first wrote my comment, so no change needed.